zaza/utilities/juju.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2018 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import yaml
+import zaza.model
+
+
+def _get_unit_names(model, names):
+    """
+    Helper function that resolves application names to first unit name of
+    said application.  Any already resolved unit names are returned as-is.
+
+    :param model: Juju model to run in
+    :type model: str
+    :param units: List of units/applications to translate
+    :returns: List of units
+    :rtype: list
+    """
+    result = []
+    for name in names:
+        if '/' in name:
+            result.append(name)
+        else:
+            result.append(zaza.model.get_first_unit_name(model, name))
+    return result
+
+
+def get_relation_from_unit(model, unit, remote_unit, relation_name):
+    """
+    Get relation data for relation with `relation_name` between `unit` and
+    `remote_unit` from the perspective of `unit`.
+
+    :param model: Juju model to run in
+    :type model: str
+    :param unit: Unit to get relation data from
+    :type unit: str
+    :param remote_unit: Other end of the relation we want to query
+    :type remote_unit: str
+    :param relation_name: Name of relation to query
+    :type relation_name: str
+    :returns: dict with relation data
+    :rtype: dict
+    """
+    (_unit, _remote_unit) = _get_unit_names(model, [unit, remote_unit])
+    return yaml.load(
+        zaza.model.run_on_unit(
+            model, _unit,
+            'relation-get --format=yaml -r "$(relation-ids {})" - {}'
```

Comment:
This may not work if relation-ids returns more than one id.

Reply:
Good point, thx, addressed.